app/src/main/java/com/hewgill/android/nzsldict/SignVideoFragment.java

```diff
@@ -108,10 +118,19 @@ public boolean onError(MediaPlayer mp, int what, int extra) {
                 return false;
             }
         });
+
+        final boolean[] isPrepared = { false };
```

Comment:
I can see below that you are using this to not reset multiple times, but can you explain more why it needs to be an array? I can't see that anything is being done with anything other than the first array element - can we not just `final boolean isPrepared = false`. I realise I'm probably missing something.

Reply:
because you cannot modify the value of `final` variables, so this is a common workaround where we use a variable that holds a (static) reference to an instance of a container whose value can be changed

Reply:
Does it need to be `final`, if the value is being changed in `onPrepared`? Or does it need to be final to be called inside that method?

Reply:
The latter - it's a requirement of Java that variables used in this sort of manner be (effectively) `final` (i.e. either being marked explicitly as `final` or never reassigned which Java supports inferring as "effectively final").

Reply:
Yeah, I can see that. Without `final`, it wouldn't have the same guarantees of not changing.

Reply:
Although I suspect the workaround here would have the same issue 😆